services/QuillLMS/app/views/cms/skills_reports/index.html.erb

```diff
@@ -0,0 +1,100 @@
+      <%= select_tag :concept_id,
+        options_from_collection_for_select(@concepts, :id, :name, params[:concept_id]) %>
+    </div>
+
+    <div>
+      <%= label_tag :sub_concept_id, 'Skill Sub-Category (Level 1 Concept)' %>
+      <%= select_tag :sub_concept_id,
+        options_from_collection_for_select(@sub_concepts, :id, :name, params[:sub_concept_id]),
+        include_blank: '(All Sub-Categories)' %>
+    </div>
+
+    <div>
+      <%= label_tag :smart_group_target, 'Smart Group Target Score' %>
+      <%= text_field_tag :smart_group_target, @smart_group_target %>
+    </div>
+
+    <div>
+      <%= label_tag :all_history, 'Show All Practice History (not just from this classroom)' %>
+      <%= check_box_tag :all_history, 1, @all_history %>
+    </div>
+
+    <div>
+      <%= submit_tag 'run report' %>
+      <span>Time to generate: <%= (Time.current - @start_time).round(2) %> seconds (note that this may be up to half a second shorter than the actual experienced load time)</span>
+    </div>
+  <% end %>
+
+
+
+  <% if @results.present? || params[:concept_id].present? %>
+    <h2>Report Results</h2>
+    <table class='table'>
+      <thead>
+        <tr>
+          <th>Name</th>
+          <th>Momentum Score</th>
+          <th>Previous Momentum Score</th>
+          <th>Trend</th>
+          <th>Exposure</th>
+          <th>Unique Questions Answered</th>
+          <th>Questions Replayed</th>
+          <th>Total Questions Answered</th>
+          <th>Replay Rate</th>
+          <th>Last Practiced</th>
+          <th>Smart Group</th>
+          <th>Scores</th>
+          <th>Time Spent</th>
+        </tr>
+      </thead>
+      <tbody>
+        <% @results.each do |result| %>
+          <tr>
+            <td><%= result.name %></td>
+            <td><%= result.ema %></td>
+            <td><%= result&.ema_previous %></td>
+            <td><%= result.trend == 0 || result.ema_previous.nil? ? 'Steady' : result.trend == 1 ? 'Trending Up' : 'Trending Down'   %></td>
+            <td><%= result.exposure %></td>
+            <td><%= result.unique_answered %></td>
+            <td><%= result.replays_answered %></td>
+            <td><%= result.total_answered %></td>
+            <td><%= number_to_percentage(result.replay_rate, precision: 0) %></td>
+            <td><%= result.last_practiced ? result.last_practiced.strftime('%b %e, %Y') : '-' %></td>
+            <td>
+              <% if result.exposure == 'Low' %>
+                Needs More Practice
+              <% elsif result.ema >= @smart_group_target %>
+                Meeting Goal
+              <% elsif result.ema >= @smart_group_target - 10 %>
+                Approaching Goal
+              <% elsif result.ema >= @smart_group_target - 20 %>
+                Needs Support
+              <% end %>
+            </td>
+            <td><%= result.scores.join(', ') %></td>
+            <td><%= result.timespent / 3600 %>h <%= result.timespent % 3600 / 60 %>m</td>
```

Comment:
Consider making the constants here floats to make sure you avoid the weirdness around `int / int` in ruby, e.g. make the denominators `3600.0` and `60.0`